src/meta/client/MetaClient.cpp

```diff
@@ -107,6 +110,35 @@ void MetaClient::stop() {
     isRunning_ = false;
 }
 
+void MetaClient::dnsQUeryThreadFunc() {
+    SCOPE_EXIT {
+        bgThread_->addDelayTask(
+            FLAGS_meta_domain_ttl_secs * 1000, &MetaClient::dnsQUeryThreadFunc, this);
+    };
+    // we need update addr from dns.
+    std::vector<network::InetAddress> addrs;
+    std::vector<int> addrsIndex;
+    for (auto it = addrs.cbegin(); it != addrs.cend(); ++it) {
+        if (it->isResolved()) {
+            try {
+                auto newAddr = network::InetAddress(it->getHostStr(), it->getPort(), true);
+                if (newAddr != (*it)) {
+                    addrs.emplace_back(newAddr);
+                    addrsIndex.emplace_back(std::distance(addrs.cbegin(), it));
+                }
+            } catch (std::exception& e) {
+                LOG(ERROR) << "update meta addr failed, err: " << e.what();
+            }
+            continue;
+        }
+    }
```

Comment:
Still a little confused about this code block. I suppose that what we need to do is try to resolve each meta hostname, and check the corresponding ip has updated or not.

Just like `bright-starry-sky  said, the `addrs` is empty, how can we update them? I guess we need to iterate `addrs_`, not `addrs`.

Reply:
yes, i have fixed this.